py/objmodule.c

```diff
@@ -239,45 +239,47 @@ STATIC const mp_rom_map_elem_t mp_builtin_module_table[] = {
 
 MP_DEFINE_CONST_MAP(mp_builtin_module_map, mp_builtin_module_table);
 
-// returns MP_OBJ_NULL if not found
-mp_obj_t mp_module_get(qstr module_name) {
+// Returns an already-loaded module (of any type), or MP_OBJ_NULL otherwise.
+mp_obj_t mp_module_get_loaded(qstr module_name) {
     mp_map_t *mp_loaded_modules_map = &MP_STATE_VM(mp_loaded_modules_dict).map;
     // lookup module
     mp_map_elem_t *el = mp_map_lookup(mp_loaded_modules_map, MP_OBJ_NEW_QSTR(module_name), MP_MAP_LOOKUP);
+    return el ? el->value : MP_OBJ_NULL;
+}
 
+// Returns an already-loaded module, a newly-initialised built-in, or MP_OBJ_NULL otherwise.
+mp_obj_t mp_module_get(qstr module_name) {
```

Comment:
To decrease code size it might be possible to combine `mp_module_get()` and `mp_module_get_loaded()` in to 1 function, with an argument `bool search_builtins`.

Reply:
Turns out with some more refactoring I don't need both -- can always use the "include builtins" version.

I dug into this much further got the code size down, and made an effort to make the import code a lot easier to follow. Overall PR is now -80 bytes on PYBV11. (i.e. -152 for the first commit, +72 for this feature).